src/cfnlint/rules/resources/properties/Properties.py

```diff
@@ -177,10 +177,11 @@ def check_exceptions(self, parenttype, proptype, text):
             - Start with handling exceptions for templated code.
         """
         templated_exceptions = {
-            'AWS::ApiGateway::RestApi': ['BodyS3Location'],
+            'AWS::ApiGateway::RestApi': ['S3Location'],
             'AWS::Lambda::Function': ['Code'],
             'AWS::Lambda::LayerVersion': ['Content'],
             'AWS::ElasticBeanstalk::ApplicationVersion': ['SourceBundle'],
+            'AWS::StepFunctions::StateMachine': ['S3Location'],
```

Comment:
And this [DefinitionS3Location](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-stepfunctions-statemachine.html#cfn-stepfunctions-statemachine-definitions3location) or [DefinitionS3Location](https://github.com/aws/aws-cli/blob/master/awscli/customizations/cloudformation/artifact_exporter.py#L422)

Reply:
![Screen Shot 2020-09-08 at 10 54 01 AM](https://user-images.githubusercontent.com/7014355/92511104-b708a280-f1c1-11ea-90e7-e54974f22752.png)

Reply:
As @asjohnston-asf caught here: ["I don't appreciate any nuance of the property type being S3Location vs DefinitionS3Location"](https://github.com/aws-cloudformation/cfn-python-lint/issues/1683#issuecomment-687662932), the name of the property in the template is actually different than the name of the property type in the [Resource Specification](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html)

From testing, these rules seems to be parsing the name of the property differently because of this, hence the divergence between `templated_exceptions` in the two rules